### ISSUE TITLE
Replace indexeddb for RN build

### ISSUE THREAD

**Issue by michaelsena** [3box/3box-js#443]

**Comment by Schwartz10** [comment]
This ticket is potentially relevant to using 3box in an iframe with sandbox restrictions (like aragon applications). The iframe sandbox restrictions in aragon applications prevent indexeddb from working, so orbitdb and 3box aren't immediately usable. 

What's the plan for RN?

**Comment by zachferland** [comment]
@Schwartz10 its not to difficult to replace the data layers for IPFS (ipfs-repo) and orbitdb (orbit-db-cache) with other data layers (both by default in web use indexeddb). So for RN both those interfaces will be implemented to use AsyncStorage. 

Can you use localstorage in the iframe? because it possible to map storage to that, its just much less performant

**Comment by Schwartz10** [comment]
@zachferland thanks for the quick response! This is really helpful to know. `localStorage` seems to be available in the iframe (just ran a quick test on my own, and can persist a key value pair there). 

>  its not to difficult to replace the data layers for IPFS (ipfs-repo) and orbitdb (orbit-db-cache) with other data layers

Do you happen to have any examples?

**Comment by zachferland** [comment]
yeah, here is https://github.com/3box/orbit-db-cache-redis which uses Redis for local orbitdb data, it just implements this interface https://github.com/orbitdb/orbit-db-cache. 

And then ipfs I think has a few examples, but here is one we have used https://github.com/ipfs/js-datastore-s3 and it just implements the ipfs-repo interface. 

Also both (default orbit-db-store and ipfs-repo) use abstract-leveldown compliant store interfaces within, so you may find a lib for other stores that implement that ( think there is one for local storage) and you can write even less code by passing that in.
